state.go

```diff
@@ -515,12 +527,78 @@ func (u *UpdateInstallState) Handle(ctx *StateContext, c Controller) (State, boo
 
 	if err := c.InstallUpdate(u.imagein, u.size); err != nil {
 		log.Errorf("update install failed: %s", err)
-		return NewUpdateErrorState(NewTransientError(err), u.update), false
+		return NewFetchInstallRetryState(u, u.update, err), false
 	}
 
 	return NewRebootState(u.update), false
 }
 
+type FetchInstallRetryState struct {
+	CancellableState
+
+	from   State
+	update client.UpdateResponse
+	err    error
+}
+
+func NewFetchInstallRetryState(from State, update client.UpdateResponse,
+	err error) State {
+	return &FetchInstallRetryState{
+		CancellableState: NewCancellableState(BaseState{
+			id: MenderStateCheckWait,
```

Comment:
Yes, the states were confusing to me. Should it just be a `MenderStateFetchInstallRetryWait` state?